lib/Dialect/FIRRTL/FIRRTLIntrinsics.cpp

```diff
@@ -25,6 +25,20 @@ ParseResult GenericIntrinsic::hasNInputs(unsigned n) {
   return success();
 }
 
+// Checks for a number of operands between n and m (allows for (n-m) optional
+// parameters)
+ParseResult GenericIntrinsic::hasNInputs(unsigned n, unsigned m) {
```

Comment:
Please give a better name.  the behavior is not has n inputs.

Reply:
Also, isn't this something which is already handled by variadic constraints in td?

Reply:
yes but it wasn't supported for generic intrinsics. This adds support for that. I can name it something like `hasNToMInputs`

Reply:
I gave it the same signature as `hasNParams`